app/block-sdk/signer_extraction_adapter_test.go

```diff
@@ -0,0 +1,181 @@
+package blocksdk_test
+
+import (
+	"math/big"
+	"testing"
+	"time"
+
+	sdkmath "cosmossdk.io/math"
+	"github.com/cosmos/cosmos-sdk/client"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/evmos/evmos/v14/app"
+	blocksdk "github.com/evmos/evmos/v14/app/block-sdk"
+	"github.com/evmos/evmos/v14/crypto/ethsecp256k1"
+	"github.com/evmos/evmos/v14/encoding"
+	"github.com/evmos/evmos/v14/testutil"
+	testutiltx "github.com/evmos/evmos/v14/testutil/tx"
+	evmostypes "github.com/evmos/evmos/v14/types"
+	evmtypes "github.com/evmos/evmos/v14/x/evm/types"
+	"github.com/stretchr/testify/suite"
+)
+
+type SignerExtractionAdapterTestSuite struct {
+	suite.Suite
+
+	txConfig client.TxConfig
+	pk       cryptotypes.PrivKey
+	app      *app.Evmos
+	ctx      sdk.Context
+
+	extractor blocksdk.SignerExtractionAdapter
+	signer    sdk.AccAddress
+	account   *evmostypes.EthAccount
+}
+
+const (
+	privKey = "b71c71a67e1177ad4e901695e1b4b9ee17ae16c6668d313eac2f96dbcda3f291"
+	chainID = "evmos_6102-1"
+)
+
+func TestSignerExtractionAdapterTestSuite(t *testing.T) {
+	suite.Run(t, new(SignerExtractionAdapterTestSuite))
+}
+
+func (s *SignerExtractionAdapterTestSuite) SetupTest() {
+	// create the encoding config
+	s.txConfig = encoding.MakeConfig(app.ModuleBasics).TxConfig
+	// create the app
+	s.app = app.Setup(true, nil, chainID)
+
+	// create a private key (eth)
+	ecdsaPriv, err := crypto.HexToECDSA(privKey)
+	s.Require().NoError(err)
+	s.pk = &ethsecp256k1.PrivKey{
+		Key: crypto.FromECDSA(ecdsaPriv),
+	}
```

Comment:
Nit: can refactor here to use the `NewAddrKey` function from `github.com/evmos/evmos/v14/testutil`